base/Enums.jl

```diff
@@ -137,6 +137,7 @@ macro enum(T, syms...)
         Enums.basetype(::Type{$(esc(typename))}) = $(esc(basetype))
         Base.typemin(x::Type{$(esc(typename))}) = $(esc(typename))($lo)
         Base.typemax(x::Type{$(esc(typename))}) = $(esc(typename))($hi)
+        Base.hash(x::$(esc(typename)), h::UInt64) = hash($(hash(typename)), hash(Integer(x), h))
```

Comment:
Can `h` be a `UInt32` on 32-bit systems?

Reply:
Yes it should be `UInt`.

Reply:
:+1: delaying the computation of the hash for the type until after the macro is expanded and the type is defined is nice.